recruit/jobopenings.go

```diff
@@ -0,0 +1,456 @@
+		Mobile                      string      `json:"Mobile"`
+		LastName                    string      `json:"Last_Name"`
+		CurrentSalary               interface{} `json:"Current_Salary"`
+		AssociatedAnySocialProfiles interface{} `json:"Associated_any_Social_Profiles"`
+		Fax                         interface{} `json:"Fax"`
+		ExpectedSalary              interface{} `json:"Expected_Salary"`
+	} `json:"data"`
+	Info PageInfo `json:"info,omitempty"`
+}
+
+// ============================== ABANDONED DUE TO ZOHO API v1 not properly working ==============================
+// https://help.zoho.com/portal/en/kb/recruit/developer-guide/api-methods/articles/getsearchrecords
+func (c *API) XMLSearchJobOpenings(params map[string]zohoutils.Parameter) (data JobOpeningsResponse, err error) {
+	endpoint := zohoutils.Endpoint{
+		Name:         "jobopenings",
+		URL:          fmt.Sprintf("https://recruit.zoho.%s/recruit/private/xml/%s/getSearchRecords", c.ZohoTLD, JobOpeningsModule),
+		Method:       zohoutils.HTTPGet,
+		ResponseData: &XMLJobOpeningsResponse{},
+		URLParameters: map[string]zohoutils.Parameter{
+			"fromIndex":       "1",     // Integer | Default value - 1
+			"toIndex":         "200",   // Integer | Default value - 20 | Maximum value - 200
+			"version":         "2",     // This will fetch responses based on the latest API implementation.
+			"newFormat":       "1",     // 1 - To exclude fields with "null" values while inserting data from your Recruit account.
+			"selectColumns":   "(ALL)", // mandatory eg: (Job Description) | Module(Job Description)
+			"searchCondition": "",      // mandatory eg: (Associated Tags|=|JobIsPublished)
+		},
+	}
+
+	if len(params) > 0 {
+		for k, v := range params {
+			endpoint.URLParameters[k] = v
+		}
+	}
+
+	// log.Printf("%s\n", endpoint)
+
+	err = c.Zoho.HTTPRequest(&endpoint)
+	if err != nil {
+		return JobOpeningsResponse{}, fmt.Errorf("failed to retrieve XML searched %s: %s", JobOpeningsModule, err.Error())
+	}
+
+	if v, ok := endpoint.ResponseData.(*XMLJobOpeningsResponse); ok {
+		// Convert to Json
+		total, _ := strconv.Atoi(v.Result.JobOpenings.Row.No)
+		dataXML := v.Result.JobOpenings.Row.FL
+
+		res := &JobOpeningsResponse{}
+		items := make([]*JobOpening, 0)
+
+		for i := 0; i <= total; i++ {
+			switch dataXML[i].Val {
+			case "JOBOPENINGID":
+				items = append(items, &JobOpening{ID: dataXML[i].Text})
+			case "CLIENTID":
+				items = append(items, &JobOpening{})
+			case "Client Name":
+				items = append(items, &JobOpening{})
+			case "Posting Title":
+				items = append(items, &JobOpening{JobOpeningName: dataXML[i].Text})
+			case "Job Opening ID":
+				items = append(items, &JobOpening{JobOpeningID: dataXML[i].Text})
+			case "Job Description":
+				items = append(items, &JobOpening{JobDescription: dataXML[i].Text})
+			}
+		}
+
+		res.Data = items
+
+		return *res, nil
+	}
+
+	return JobOpeningsResponse{}, fmt.Errorf("no 'JobOpeningsResponse' returned")
+}
```

Comment:
That's unfortunate! You've clearly put alot of time and attention into this PR, I really appreciate it.